File: src/main/java/com/kerimsenturk/labreport/dto/request/CreateUserRequest.java
package com.kerimsenturk.labreport.dto.request;

import com.kerimsenturk.labreport.model.enums.UserRole;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import org.hibernate.validator.constraints.Length;

public record CreateUserRequest(
        @NotBlank
        @Length(min = 7, max = 11) //userId has 7 digit for hospital personals, 11 digit for patients.
        @Pattern(regexp = "[0-9]+")
        String userId,

        @NotBlank
        String name,

        @NotBlank
        String surname,

        @NotBlank
        @Length(min = 7, max = 20)
        //TODO: Pattern working but it can not accept any string. I think it is about UTF format. Fix it
        //@Pattern(regexp = "(?=.*?[a-z]+)(?=.*?[A-Z]+)(?=.*?[0-9]+)")
        String password,

        //It may be necessary to create a new EnumValidator Annotation
        UserRole userRole) {

}

File: src/main/web-ui/src/domain/payload/request/CreateUserRequest.ts
import {UserRole} from "../../model/User";

class CreateUserRequest {
    userId: string;

    name: string;

    surname: string;

    password: string;

    userRole: UserRole;

    constructor(userId: string, name: string, surname: string, password: string, userRole: UserRole) {
        this.userId = userId;
        this.name = name;
        this.surname = surname;
        this.password = password;
        this.userRole = userRole;
    }
}
